### ISSUE TITLE
can't getwinposx() over ssh because it's slow

### ISSUE THREAD

**Issue by lilydjwg** [vim/vim#2689]
I find that `getwinposx()` returns `-1` when I ssh into a machine (in xterm of course) and run Vim. I do this because I'm trying to see if the window is on my E-ink screen so I can adjust a few things.

Sadly in `term_get_winpos` Vim only gives the terminal 100ms to response, in which the light speed is still too slow for the data to arrive.

Can Vim wait longer? Or make it adjustable via an option?

**Comment by lilydjwg** [comment]
Thanks, it works great!

**Comment by chdiza** [comment]
I don't see why it was necessary to add a function for this.  What was wrong with the command `:winpos`?  It's easily parseable in VimScript.

**Comment by lilydjwg** [comment]
I forgot about `:winpos`, but it doesn't work in a terminal (maybe we should make it work). Also, it's not very easy to parse because of translations.

**Comment by chdiza** [comment]
You're right it doesn't work in a terminal.  I forgot about that part.

But I don't see why translations would make it hard to parse.  All you need are the second and fourth substrings, where "space" is the separator.  Then trim the comma from one of them.

**Comment by lilydjwg** [comment]
>  All you need are the second and fourth substrings, where "space" is the separator.

No, there are "Window position:" preceding, so it's 4th and 6th in English, 3rd and 5th in Chinese, and 7th and 9th in French. Counting backwards may be better, but it feel like a trick.

**Comment by chdiza** [comment]
I meant skipping over the colon, of course :)
